### PR TITLE
Conversation settings enhancements

### PR DESCRIPTION
**Pull request by Ivansss**
Display conversation settings as in the Web UI

IMG_0099 IMG_0100

### CODE REVIEW

**Comment by SystemKeeper** [comment]
In Web "Leave Conversation" is not displayed red, should this be inline with web?
image

Maybe a different icon for both Delete-Options might make sense? 
https://fonts.google.com/icons?icon.query=delete
(I can see "Delete Forever" work for "Delete Conversation" or "Delete Sweep" for "Delete all messages")

**Comment by Ivansss** [comment]
> In Web "Leave Conversation" is not displayed red, should this be inline with web?

I also thought about that. The thing is that "Leave conversation" is a destructive action if the conversation is not an "open conversation". So we either add an explanation (which will be a bit long for mobiles) or we leave it red for not opened coversations and gray for open ones.

> Maybe a different icon for both Delete-Options might make sense?
> https://fonts.google.com/icons?icon.query=delete
> (I can see "Delete Forever" work for "Delete Conversation" or "Delete Sweep" for "Delete all messages")

I also don't really like to have same icons next to each other for 2 different actions.
The icons you suggested I guess they would look good :)

What do you think @marcoambrosini @jancborchardt ?

File: NextcloudTalk/RoomInfoTableViewController.m [hunk 1]
```diff
@@ -1644,19 +1598,19 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
...
         {
-            NSArray *actions = [self getPublicActions];
-            PublicAction action = [[actions objectAtIndex:indexPath.row] intValue];
+            NSArray *actions = [self getGuestsActions];
+            GuestAction action = [[actions objectAtIndex:indexPath.row] intValue];
             switch (action) {
-                case kPublicActionPublicToggle:
+                case kGuestActionPublicToggle:
                 {
                     UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:shareLinkCellIdentifier];
```
Comment (SystemKeeper) [inline NextcloudTalk/RoomInfoTableViewController.m]: Rename to `allowGuestsCellIdentifier` or similar ?

File: NextcloudTalk/RoomInfoTableViewController.m [hunk 2]
```diff
@@ -1536,65 +1516,39 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
...
+        case kRoomInfoSectionNotifications:
         {
-            NSArray *actions = [self getRoomActions];
-            RoomAction action = [[actions objectAtIndex:indexPath.row] intValue];
+            NSArray *actions = [self getNotificationsActions];
+            NotificationAction action = [[actions objectAtIndex:indexPath.row] intValue];
             switch (action) {
-                case kRoomActionFavorite:
-                {
-                    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:favoriteRoomCellIdentifier];
```
Comment (SystemKeeper) [inline NextcloudTalk/RoomInfoTableViewController.m:1546]: `favoriteRoomCellIdentifier` is still defined

Comment (SystemKeeper) [reply NextcloudTalk/RoomInfoTableViewController.m:1546]: Also `kModificationErrorFavorite` is there as well.

File: NextcloudTalk/RoomInfoTableViewController.m (hunk 1 above)
Comment (Ivansss) [reply NextcloudTalk/RoomInfoTableViewController.m]: Fixed!

File: NextcloudTalk/RoomInfoTableViewController.m (hunk 2 above)
Comment (Ivansss) [reply NextcloudTalk/RoomInfoTableViewController.m:1546]: Fixed!

**Comment by Ivansss** [comment]
This is the final look for the destructive actions

IMG_E0102

**Comment by marcoambrosini** [comment]
I would go non-red with the explanation like in the web view for the leave conversation. I wouldn't worry too much about space as our scrolling muscles are pretty well trained nowadays and those are not very frequent actions anyway :) 

As per the action icons, do you have something similar to this for the delete messages? https://materialdesignicons.com/icon/delete-clock

**Comment by Ivansss** [comment]
> As per the action icons, do you have something similar to this for the delete messages? https://materialdesignicons.com/icon/delete-clock

Do you mean for "delete all messages" action or for delete a single message in the chat view?

**Comment by Ivansss** [comment]
Added a new icon for "Delete all messages" icon
https://materialdesignicons.com/icon/chat-remove-outline
Thanks @SystemKeeper for the finding :)

IMG_E0103

Also added information about "Leave conversation" in the confirmation dialog

IMG_E0104

**Comment by marcoambrosini** [comment]
Looks good!

**Pull request merged**